### PR TITLE
eth/gasestimator: include blobs in virtual balance computation

### PR DESCRIPTION
**Pull request by nand2**
Fix for the bug descripted on https://github.com/ethereum/go-ethereum/issues/29702 -- I no longer see the bug.

I am new to go-ethereum codebase, so I may have missed things, please tell me :-)

Also, would you like some test with it?

### CODE REVIEW

File: eth/gasestimator/gasestimator.go [hunk 1]
```diff
@@ -80,6 +80,11 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 			}
 			available.Sub(available, call.Value)
 		}
+		if opts.Config.IsCancun(opts.Header.Number, opts.Header.Time) && len(call.BlobHashes) > 0 {
+			blobBalanceUsage := new(big.Int).SetInt64(int64(len(call.BlobHashes) * params.BlobTxBlobGasPerBlob))
+			blobBalanceUsage.Mul(blobBalanceUsage, call.BlobGasFeeCap)
+			available.Sub(available, blobBalanceUsage)
```
Comment (fjl) [inline eth/gasestimator/gasestimator.go:86]: I think we need another check for `balance > blobBalanceUsage` here, otherwise it could go negative.

Comment (nand2) [reply eth/gasestimator/gasestimator.go:86]: You're right! Which one of the following options would you use for the error?

- Using `core.ErrInsufficientFunds` as error, keeping `insufficient funds for gas * price + value` as description,
-- Using `core.ErrInsufficientFunds` as error, changing description to  `insufficient funds for gas * price + blobGas * blobGasPrice + value` as description,
- creating and using a new error, `ErrEip4844InsufficientFunds` with `insufficient funds for gas * price + blobGas * blobGasPrice + value` as description ?

I guess option 3?

Comment (nand2) [reply eth/gasestimator/gasestimator.go:89]: Added option 1/ for now, let me know.

**Review by holiman — APPROVED** [review]
LGTM

**Review by fjl — APPROVED** [review]

**Pull request merged**